jbpm-system-test/jbpm-unmarshall-ejb-timers-test/src/test/java/org/kie/samples/integration/UnmarshallEJBTimerSystemTest.java

```diff
@@ -0,0 +1,220 @@
+    
+    private static KieServicesClient ksClient1;
+    private static KieServicesClient ksClient2;
+    
+    private static ProcessServicesClient processClient1;
+    private static ProcessServicesClient processClient2;
+    
+    private static HikariDataSource ds;
+    
+    @BeforeClass
+    public static void setup() {
+        logger.info("KIE SERVER 1 started at port "+kieServer1.getKiePort());
+        logger.info("KIE SERVER 2 started at port "+kieServer2.getKiePort());
+        logger.info("postgresql started at "+postgreSQLContainer.getJdbcUrl());
+        
+        ksClient1 = authenticate(kieServer1.getKiePort(), DEFAULT_USER, DEFAULT_PASSWORD);
+        ksClient2 = authenticate(kieServer2.getKiePort(), DEFAULT_USER, DEFAULT_PASSWORD);
+        
+        processClient1 = ksClient1.getServicesClient(ProcessServicesClient.class);
+        processClient2 = ksClient2.getServicesClient(ProcessServicesClient.class);
+        
+        createContainer(ksClient1);
+        createContainer(ksClient2);
+        
+        ds = getDataSource();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        ksClient1.disposeContainer(containerId);
+        ksClient2.disposeContainer(containerId);
+        DockerClient docker = DockerClientFactory.instance().client();
+        docker.listImagesCmd().withLabelFilter("autodelete=true").exec().stream()
+         .filter(c -> c.getId() != null)
+         .forEach(c -> docker.removeImageCmd(c.getId()).withForce(true).exec());
+    }
+
+    @Test
+    @DisplayName("user starts process in one node but get variable in another before refresh-time")
+    public void getVariableBeforeRefresh() throws Exception {
+        Long processInstanceId = startProcessInNode1("ht-timer-process");
+        
+        Thread.sleep(2000);
+        
+        findVariableInNode2(processInstanceId);
+        
+        assertEquals("there should be just one timer at the table",
+                      1, performQuery(SELECT_COUNT_FROM_JBOSS_EJB_TIMER).getInt(1));
+        
+        assertEquals("timer should be started at node1 partition",
+                     "ejb_timer_node1_part", performQuery(SELECT_PARTITION_NAME_FROM_JBOSS_EJB_TIMER).getString(1));
+        
+    }
+
+    private static void createContainer(KieServicesClient client) {
+        ReleaseId releaseId = new ReleaseId(GROUP_ID, ARTIFACT_ID, VERSION);
+        KieContainerResource resource = new KieContainerResource(containerId, releaseId);
+        resource.setContainerAlias(ARTIFACT_ID + ALIAS);
+        client.createContainer(containerId, resource);
+    }
+
+    private static void createCLIFile(String nodeName) {
+        Boolean noCluster = Boolean.getBoolean("org.kie.samples.ejbtimer.nocluster");
+        //if different partitions are defined per nodeName, then there is no cluster for EJB timers
+        String node = noCluster? nodeName : "node1";
```

Comment:
we're not defining this system property anywhere in the test - do we really need it or we can just keep it as is?

Reply:
we can keep it as it is, in case "no cluster" scenario could be also tested